### PR TITLE
Basic Landing Page of Website

### PR DESCRIPTION
**Pull request by Prashantneu8848**
This PR entails the following changes:
1. Contains basic landing page for the website.
2. Contains a fix Navigation Bar at the top.
3. Contains React-bootstrap component (Navigation Bar).
4. Tests for the changes.
5. jest.config.js file to ignore css.
6. Contains bootstrap.
7. Created components directory inside /src and moved ReceiptInput.js and ReceiptInput.test.js inside it.

screenshot: https://drive.google.com/file/d/1poU3uJ7kySx6pFDbiiXkVbWO_jXzL6dV/view?usp=sharing

### CODE REVIEW

**Review by livseibert5 — APPROVED** [review]

File: edith-step/src/index.css [hunk 1]
```diff
@@ -1,13 +1,4 @@
...
-  -moz-osx-font-smoothing: grayscale;
-}
-
-code {
-  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
-    monospace;
+body { 
+  height: 100vh;
+  width: 100vw;
 }
```
Comment (malachibre) [inline edith-step/src/index.css:4]: Nit: Add new line.

File: edith-step/src/App.css [hunk 2]
```diff
@@ -1,38 +1,51 @@
...
+@keyframes scroll-down {
+  0% {
+    margin-top:0px;
   }
-  to {
-    transform: rotate(360deg);
+  100% {
+    margin-top:15px;
   }
 }
```
Comment (malachibre) [inline edith-step/src/App.css:51]: Nit: Add new line. (Also add new lines in App.js, App.test.js, ReceiptInput.test.js, TopNavBar.js, TopNavBar.test.js, and index.css)

**Review by malachibre — APPROVED** [review]

File: edith-step/src/App.css (hunk 2 above)
Comment (Prashantneu8848) [reply edith-step/src/App.css:51]: Seems like there are new lines in those files. Locally  it shows there are new lines at the end of the file.

File: edith-step/src/App.test.js [hunk 3]
```diff
@@ -1,9 +1,24 @@
...
+})
+
+afterEach(() => {
+  component.unmount();
+});
+
+describe('App component', () => {
+  // Checks if the app is rendered.
+  test('renders', () => {
+
```
[the remark this answers is not in the document]
Comment (Prashantneu8848) [reply edith-step/src/App.test.js:21]: Changes made. Thanks!

File: edith-step/src/App.js [hunk 4]
```diff
@@ -1,15 +1,31 @@
...
 
+import ReceiptInput from './components/ReceiptInput';
+import TopNavbar from './components/TopNavbar';
 
-/** Main webpage for the website. */
+import './App.css';
+
+/**
+ * Main webpage for the website.
+ */
```
Comment (sehtia-raghav) [inline edith-step/src/App.js:10]: nit: I think we can make this an inline javadoc

`/** Main webpage for the website. */`

File: edith-step/src/components/TopNavbar.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+import Navbar from 'react-bootstrap/Navbar';
+import Nav from 'react-bootstrap/Nav';
+
+/**
+ * Top navigation bar for user interaction with loggin, loggin out
```
Comment (sehtia-raghav) [inline edith-step/src/components/TopNavbar.js:7]: nit: "log-in" ?

**Review by sehtia-raghav — CHANGES_REQUESTED** [review]

File: edith-step/src/App.js [hunk 6]
```diff
@@ -1,15 +1,31 @@
...
 class App extends Component {
-  /** Renders the main webpage. */
+
   render() {
```
Comment (corycook) [inline edith-step/src/App.js:13]: Lint error: Missing JSDoc comment.

Please fix.

File: edith-step/src/App.js [hunk 7]
```diff
@@ -1,15 +1,31 @@
...
 
-/** Main webpage for the website. */
+import './App.css';
+
+/**
+ * Main webpage for the website.
+ */
 class App extends Component {
-  /** Renders the main webpage. */
+
```
Comment (corycook) [inline edith-step/src/App.js:12]: Lint error: "Block must not be padded by blank lines"

Please fix.

File: edith-step/src/App.js [hunk 8]
```diff
@@ -1,15 +1,31 @@
...
+        <div className='App-header'>
+          <TopNavbar />
+        </div>
+        <div className='background-image-0'>
+          <div className='scroll-down border'>
+            Scroll Down
+          </div>
+          <div className='app-describe'>
+            <span className='border'>
+              Welcome To Edith: The Best Expenditure Analyzer.
```
Comment (corycook) [inline edith-step/src/App.js:25]: Avoiding lofty claims will make the app seem more authentic. Maybe just "Welcome to Edith Expenditure Analyzer" would be better.

File: edith-step/src/App.js [hunk 9]
```diff
@@ -1,15 +1,31 @@
...
+            </span>
+          </div>
+        </div>
         <ReceiptInput />
```
Comment (corycook) [inline edith-step/src/App.js:29]: Should probably move this to its own view and implement some kind of React navigation in a future PR.

No action necessary.

File: edith-step/src/components/TopNavbar.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+ * and other app features.
+ */
+class TopNavbar extends React.Component {
+
```
Comment (corycook) [inline edith-step/src/components/TopNavbar.js:11]: Lint error: "Block must not be padded by blank lines"

Please fix.

File: edith-step/src/components/TopNavbar.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  render() {
```
Comment (corycook) [inline edith-step/src/components/TopNavbar.js:12]: Lint error: "Missing JSDoc comment"

Please fix.

File: edith-step/src/components/TopNavbar.test.js [hunk 12]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import {mount} from 'enzyme';
+import TopNavbar from './TopNavbar';
+import '../setupTests.js';
+
+let component;
+
+beforeEach(() => {
+  component = mount(<TopNavbar />);
+})
```
Comment (corycook) [inline edith-step/src/components/TopNavbar.test.js:10]: Lint error: "Missing semicolon"

Please fix.

File: edith-step/src/index.js [hunk 13]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
+import 'bootstrap/dist/css/bootstrap.min.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
```
Comment (corycook) [inline edith-step/src/index.js:7]: Lint errors:

   9:1   error  Expected indentation of 4 spaces but found 2  indent
  10:1   error  Expected indentation of 6 spaces but found 4  indent
  11:1   error  Expected indentation of 4 spaces but found 2  indent
  12:1   error  Expected indentation of 4 spaces but found 2  indent
  12:34  error  Missing trailing comma                        comma-dangle

Please fix.

File: jest.config.js [hunk 14]
```diff
@@ -0,0 +1,8 @@
+//jest.config.js
```
Comment (corycook) [inline jest.config.js:1]: Lint errors:

  1:1   error  Expected space or tab after '//' in comment           spaced-comment
  4:3   error  Strings must use singlequote                          quotes
  5:1   error  This line has a length of 125. Maximum allowed is 80  max-len
  5:5   error  Strings must use singlequote                          quotes
  5:92  error  Strings must use singlequote                          quotes
  6:5   error  Strings must use singlequote                          quotes
  6:28  error  Strings must use singlequote                          quotes
  6:48  error  Missing trailing comma                                comma-dangle
  7:4   error  Missing trailing comma                                comma-dangle
  8:2   error  Missing semicolon                                     semi

Please fix or exclude this file from linting.

File: edith-step/src/App.css [hunk 15]
```diff
@@ -1,38 +1,52 @@
...
+  top: 10%;
+  width: 100%;
 }
 
-@keyframes App-logo-spin {
-  from {
-    transform: rotate(0deg);
+@keyframes scroll-down {
+  0% {
+    margin-top: 0px;
```
Comment (corycook) [inline edith-step/src/App.css:46]: Lint error: "Unexpected unit"

Please fix.

I think it means that if you have zero as the value then you don't need to indicate a unit.

```
    margin-top: 0;
```

File: src/main/webapp/index.html [hunk 16]
```diff
@@ -1 +1 @@
-<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><link rel="icon" href="/favicon.6937a946.ico"><meta name="viewport" content="width=device-width, initial-scale=1"><meta name="theme-color" content="#000000"><meta name="description" content="Web site created using create-react-app"><link rel="apple-touch-icon" href="/logo192.ec1737a3.png"><link rel="manifest" href="/manifest.js"><title>React App</title><link rel="stylesheet" href="/src.4674b8f8.css"></head><body> <noscript>You need to enable JavaScript to run this app.</noscript> <div id="root"></div>  <script src="/src.be084630.js"></script> </body></html>
```
Comment (corycook) [inline src/main/webapp/index.html:1]: Files in src/main/webapp should be excluded from linting.

Lint error: 
> src/main/webapp/index.html: line 1, col 622, line ending does not match format: lf

File: edith-step/src/App.css [hunk 17]
```diff
@@ -1,38 +1,52 @@
...
 
-.App-link {
-  color: #61dafb;
+.scroll-down {
+  animation-delay: 1s;
+  animation-direction: alternate;
+  animation-duration: 1s;
+  animation-iteration-count: infinite;
+  animation-name: scroll-down;
+  animation-timing-function: ease-in-out;
```
Comment (corycook) [inline edith-step/src/App.css:36]: > Use shorthand properties where possible [...] even in cases where only one value is explicitly set.

go/htmlcssstyle#Shorthand_Properties

```css
animation: 1s ease-in-out 1s infinite alternate scroll-down;
```

File: edith-step/src/App.css [hunk 18]
```diff
@@ -1,38 +1,52 @@
...
-
-.App-logo {
-  height: 40vmin;
-  pointer-events: none;
+.background-image-0 {
+  background-attachment: fixed;
+  background-image: url(./receipt.jpg);
+  background-position: center;
+  background-repeat: no-repeat;
+  background-size: cover;
```
Comment (corycook) [inline edith-step/src/App.css:6]: > Use shorthand properties where possible [...] even in cases where only one value is explicitly set.

go/htmlcssstyle#Shorthand_Properties

```css
background: fixed url(./receipt.jpg) center/cover no-repeat; 
```

**Review by corycook — CHANGES_REQUESTED** [review]
It looks like the Java code doesn't compile.

You will want all presubmit checks to be passing before this PR is submitted that way they aren't broken or useless in all subsequent PRs.

File: edith-step/src/App.css (hunk 17 above)
Comment (corycook) [reply edith-step/src/App.css:36]: Alternate URL: https://google.github.io/styleguide/htmlcssguide.html#Shorthand_Properties

See https://developer.mozilla.org/en-US/docs/Web/CSS/animation

File: edith-step/src/App.js (hunk 4 above)
Comment (Prashantneu8848) [reply edith-step/src/App.js:10]: I was inline before but I got a lint error and changed it.

File: edith-step/src/components/TopNavbar.js (hunk 5 above)
Comment (Prashantneu8848) [reply edith-step/src/components/TopNavbar.js:7]: Changed it. Thanks

File: edith-step/src/App.js (hunk 6 above)
Comment (Prashantneu8848) [reply edith-step/src/App.js:13]: Changes made.

File: edith-step/src/App.js (hunk 7 above)
Comment (Prashantneu8848) [reply edith-step/src/App.js]: Changes made.

File: edith-step/src/App.js (hunk 8 above)
Comment (Prashantneu8848) [reply edith-step/src/App.js]: Thanks for the suggestion. Changed to  "Welcome to Edith Expenditure Analyzer".

File: edith-step/src/components/TopNavbar.js (hunk 10 above)
Comment (Prashantneu8848) [reply edith-step/src/components/TopNavbar.js]: Changes made.

File: edith-step/src/App.css (hunk 15 above)
Comment (Prashantneu8848) [reply edith-step/src/App.css]: Changes made.

File: edith-step/src/App.css (hunk 17 above)
Comment (Prashantneu8848) [reply edith-step/src/App.css]: Thanks for the reference. Changes made.

File: edith-step/src/App.css (hunk 18 above)
Comment (Prashantneu8848) [reply edith-step/src/App.css]: Thanks for the edits. Changes made.

File: edith-step/src/components/TopNavbar.js (hunk 11 above)
Comment (Prashantneu8848) [reply edith-step/src/components/TopNavbar.js:15]: Changes made.

File: jest.config.js (hunk 14 above)
Comment (Prashantneu8848) [reply jest.config.js]: Added a rule to exclude the max-len lint error. Fixed other error

**Review by sehtia-raghav — APPROVED** [review]

**Review by corycook — APPROVED** [review]
LGTM

**Pull request merged**